hypothesis-python/src/hypothesis/database.py

```diff
@@ -69,7 +69,7 @@ def _db_for_path(path=None):
                 stacklevel=3,
             )
             return InMemoryExampleDatabase()
-    if path in (None, ":memory:"):
+    if path in (None, ":memory:") or not os.access(path, os.R_OK | os.W_OK | os.X_OK):
```

Comment:
I think this is wrong because the path may not exist yet.

The difficulty is that we don't want to create it before we actually need to write something either, both for performance reasons and to avoid annoying people who have Hypothesis installed but aren't using it.